lib/srv/db/common/auth.go

```diff
@@ -1023,3 +1023,28 @@ func (r *elastiCacheRedisIAMTokenRequest) getSignableRequest() (*http.Request, e
 	}
 	return req, nil
 }
+
+type reportingAuth struct {
+	Auth
+	component        string
+	databaseProtocol string
+}
+
+func GetReportingAuth(protocol string, auth Auth) Auth {
+	return &reportingAuth{
+		Auth:             auth,
+		component:        "db:auth",
+		databaseProtocol: protocol,
+	}
+}
+
+func (r *reportingAuth) GetTLSConfig(ctx context.Context, sessionCtx *Session) (*tls.Config, error) {
+	defer r.methodCallMetrics()()
+	return r.Auth.GetTLSConfig(ctx, sessionCtx)
+}
+
+func (r *reportingAuth) methodCallMetrics() func() {
+	return methodCallMetrics(r.component, r.databaseProtocol, 2)
+}
```

Comment:
The `methodCallMetrics` is too generic and referring to runtime runtime stacktrace many times seems to be bad ideal. We we cannot just forward a func name ?

Reply:
We are picking up the function name from the stacktrace, and by using the stacktrace info we don't have to manually provide the function names everywhere. Are you worried about the performance impact here? I'm pretty sure this is rather fast and on par with `trace.Wrap` which is everywhere.

Reply:
I've dropped all but 2 calls, which makes it feasible to pass the method name explicitly without runtime information, which removes the code in question.